projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -850,6 +851,26 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy {
             } else if (expand) {
                 (this.gridAPI as any).trigger_row_expansion_toggle(this.row.treeRow, !this.row.expanded, event, this.visibleColumnIndex);
             }
+        } else if ((this.grid as IgxGridComponent).hasDetails && this.isToggleKey(key)) {
+            const collapse = (this.row as any).expanded && ROW_COLLAPSE_KEYS.has(key);
+            const expand = !(this.row as any).expanded && ROW_EXPAND_KEYS.has(key);
+            const expandedStates = this.grid.expansionStates;
+            if (expand) {
+                expandedStates.set(this.row.rowID, true);
+            } else if (collapse) {
+                expandedStates.set(this.row.rowID, false);
+            }
+            this.grid.expansionStates = expandedStates;
+            this.grid.notifyChanges();
+            const isVirtualized = !this.grid.verticalScrollContainer.dc.instance.notVirtual;
+            // persist focused cell
+            const el = this.grid.selectionService.activeElement;
+            if (isVirtualized && el) {
+                const cell = this.grid.gridAPI.get_cell_by_visible_index(el.row, el.column);
+                if (cell) {
+                    cell.nativeElement.focus();
+                }
```

Comment:
This won't always persist the focus on the right cell element as `notifyChanges` is run on the next detection cycle and the layout of the grid body can shift up/down

Reply:
Related to #6191